src/account/account.repository.knex.ts

```diff
@@ -11,6 +12,28 @@ export class KnexAccountRepository {
         private readonly events: AsyncEvents,
     ) {}
 
+    async save(account: Account): Promise<void> {
+        if (account.isNew) {
+            throw new Error(
+                'Saving of new Accounts has not been implemented yet.',
```

Comment:
We can rename this method to `update` and use it only for this purpose, and have a different implementation for `save`. My thinking is that this method is updating the account and sending update event.

Reply:
Our repository pattern uses a single save even for both update and saves, this means that callers of the code don't need to know if the entity they have a reference to is new or not.

We could change this if you think it's worth it but I think it needs to be part of a larger discussion and the PostRepository would need to be updated too

Reply:
In that case, it’s fine. I get that the single `save` method simplifies things in `PostRepository`, we can keep the same patter here.